### PR TITLE
fix: allow projects with publishers to be removed

### LINKED ISSUE #15729: Removing a project with OIDC configured fails
When removing a project via admin (such as malware), currently the removal fails, presumably due to not having an on delete cascade for the foreign key relationship.

```python
(psycopg.errors.ForeignKeyViolation) update or delete on table "projects" violates foreign key constraint "oidc_publisher_project_association_project_id_fkey" on table "oidc_publisher_project_association"
DETAIL:  Key (id)=(REDACTED) is still referenced from table "oidc_publisher_project_association".
[SQL: DELETE FROM projects WHERE projects.id = %(id)s::UUID]
[parameters: {'id': UUID('REDACTED')}]
```

We probably want to delete a publisher on project removal, since it's a one-to-one relationship, is that correct @woodruffw ?

We probably also want Admin UI functionality to remove an OIDC publisher from a project.

### PR DESCRIPTION
**Pull request by miketheman**
Resolves #15729

### CODE REVIEW

**Review by woodruffw — APPROVED** [review]
LGTM! IMO we could also delete the publisher itself _if_ the deleted project is the last one associated with it, but that's minor.

(We could also just do that periodically by deleting any "orphaned" providers, i.e. those with no projects.)

File: warehouse/migrations/versions/34c3175f4bea_remove_oidc_association_on_project_.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def upgrade():
+    op.drop_constraint(
+        "oidc_publisher_project_association_project_id_fkey",
+        "oidc_publisher_project_association",
+        type_="foreignkey",
+    )
+    op.create_foreign_key(
+        None,
```
Comment (di) [inline warehouse/migrations/versions/34c3175f4bea_remove_oidc_association_on_project_.py:33]: Just curious, did you change the constraint name to be `None` or was this auto-generated? I'm just surprised it's different than when it was first added.

Comment (miketheman) [reply warehouse/migrations/versions/34c3175f4bea_remove_oidc_association_on_project_.py:33]: It's autogenerated, and I confirmed that the constraint name remains `oidc_publisher_project_association_project_id_fkey` after the migration completes.

Comment (di) [reply warehouse/migrations/versions/34c3175f4bea_remove_oidc_association_on_project_.py:33]: Weird! 🤷

**Review by di — APPROVED** [review]

**Pull request merged**